crates/sb_graph/lib.rs

```diff
@@ -199,68 +201,70 @@ pub struct ExtractEszipPayload {
     pub folder: PathBuf,
 }
 
-fn ensure_unix_relative_path(path: &Path) -> &Path {
-    assert!(path.is_relative());
-    assert!(!path.to_string_lossy().starts_with('\\'));
-    path
-}
-
-fn create_module_path(global_specifier: &str, entry_path: &Path, output_folder: &Path) -> PathBuf {
-    let cleaned_specifier = global_specifier.replace(entry_path.to_str().unwrap(), "");
-    let module_path = PathBuf::from(cleaned_specifier);
-
-    if let Some(parent) = module_path.parent() {
-        if parent.parent().is_some() {
-            let output_folder_and_mod_folder = output_folder.join(
-                parent
-                    .strip_prefix("/")
-                    .unwrap_or_else(|_| ensure_unix_relative_path(parent)),
-            );
-            if !output_folder_and_mod_folder.exists() {
-                create_dir_all(&output_folder_and_mod_folder).unwrap();
-            }
-        }
-    }
-
-    output_folder.join(
-        module_path
-            .strip_prefix("/")
-            .unwrap_or_else(|_| ensure_unix_relative_path(&module_path)),
-    )
-}
-
 async fn extract_modules(
     eszip: &EszipV2,
     specifiers: &[String],
     lowest_path: &str,
     output_folder: &Path,
 ) {
+    let closest_common_dir = closest_common_directory(specifiers);
+    let closest_common_dir_buf = normalize_path(PathBuf::from(closest_common_dir));
     let main_path = PathBuf::from(lowest_path);
     let entry_path = main_path.parent().unwrap();
+
     for global_specifier in specifiers {
-        let module_path = create_module_path(global_specifier, entry_path, output_folder);
+        let full_path = PathBuf::from(global_specifier);
+        let is_insider_path = full_path.starts_with(entry_path);
+
+        let target_folder = if is_insider_path {
+            output_folder.join("src")
```

Comment:
I don't get why it's necessary to create `src` in output directory. Since you've found the lowest common directory, can we just set that as the output directory? Because every path will be children of the lowest common directory.